presto-main/src/main/java/io/prestosql/operator/OrderByOperator.java

```diff
@@ -498,4 +539,19 @@ public boolean supportsConsolidatedWrites()
 
         private Object spiller;
     }
+
+    private void checkFutureAndSetState()
```

Comment:
shouldnt this function just check future and set the state?why we are doing business work here? like spilling